3-reveal/migrations/5-IRS/4-Zambia-2020/deploy/remove_hard_coded_jurisdiction_depth.psql

```diff
@@ -0,0 +1,1201 @@
+    COALESCE(events_query.business_status, tasks_query.business_status, 'No Tasks') AS business_status,
+    COALESCE(events_query.sprayed_values, ARRAY[]::varchar[]) AS sprayed_values,
+    COALESCE(events_query.notsprayed_reasons, ARRAY[]::varchar[]) AS notsprayed_reasons,
+    COALESCE(events_query.duplicate, FALSE) AS duplicate,
+    COALESCE(events_query.sprayed_duplicate, FALSE) AS sprayed_duplicate
+FROM locations
+INNER JOIN pending_:schema .zambia_structure_jurisdictions AS zambia_structure_jurisdictions
+    ON zambia_structure_jurisdictions.id = locations.id
+LEFT JOIN LATERAL (
+    SELECT
+        subq.task_id AS task_id,
+        subq.task_status AS task_status,
+        subq.event_id AS event_id,
+        subq.event_date AS event_date,
+        subq.plan_id AS plan_id,
+        subq.rooms_eligible AS rooms_eligible,
+        subq.rooms_sprayed AS rooms_sprayed,
+        subq.eligibility AS eligibility,
+        subq.structure_sprayed AS structure_sprayed,
+        subq.business_status AS business_status,
+        subq.notsprayed_reason AS notsprayed_reason,
+        array_agg(subq.structure_sprayed) OVER (PARTITION BY structure_sprayed) AS sprayed_values,
+        array_agg(subq.notsprayed_reason) FILTER (WHERE subq.notsprayed_reason != '') OVER (PARTITION BY notsprayed_reason) AS notsprayed_reasons,
+        array_length(array_agg(subq.structure_sprayed) OVER (PARTITION BY structure_sprayed), 1) > 1 AS duplicate,
+        'yes' = ANY (array_agg(subq.structure_sprayed) OVER (PARTITION BY structure_sprayed)) AND array_length(array_agg(subq.structure_sprayed) OVER (PARTITION BY structure_sprayed), 1) > 1 AS sprayed_duplicate
+    FROM
+    (
+        SELECT
+            DISTINCT ON (events.task_id)
+            events.id AS event_id,
+            events.task_id AS task_id,
+            events.event_date AS event_date,
+            tasks.plan_identifier AS plan_id,
+            tasks.status AS task_status,
+            -- get fields from JSON --
+            COALESCE (events.form_data->'rooms_eligible'->>0, '0')::int AS rooms_eligible,
+            COALESCE (events.form_data->'rooms_sprayed'->>0, '0')::int AS rooms_sprayed,
+            COALESCE (events.form_data->'eligibility'->>0, 'Eligible')::text AS eligibility,
+            COALESCE (events.form_data->'structure_sprayed'->>0, 'No')::text AS structure_sprayed,
+            events.form_data->>'start' AS form_start_time,
+            events.form_data->>'end' AS form_end_time,
+            events.form_data->'business_status'->>0 AS business_status,
+            COALESCE (events.form_data->'notsprayed_reason'->>0, '')::text AS notsprayed_reason
+            -- end get fields from JSON --
+        FROM events
+        LEFT JOIN tasks
+            ON tasks.identifier = events.task_id
+        WHERE locations.id = events.base_entity_id
+        AND events.entity_type = 'Structure'
+        AND events.event_type = 'Spray'
+    ) AS subq
+    ORDER BY
+        subq.form_end_time DESC,
```

Comment:
Switch to ordering by form end time which is more accurate for distinguishing between events from the same day.